File: 005 Longest Palindromic Substring/LongestPalindromicSubstring_brutalForce.java
class LongestPanlindromicSubstring_brutalForce {
    public String longestPalindrome(String s) {
        int n = s.length(); 
        if (n <= 1) {  // base case
            return s;
        }
                
        // Set<String> set = new HashSet<>();
        int max = 0;
        String res = "";
        for (int i = 0; i < n; i++) {
            for (int j = i; j < n; j++) {
                String subs = s.substring(i, j + 1);
                if (isPalindrome(subs)) {
                    if (max < subs.length()) {
                        res = subs;
                        max = subs.length();
                    }
                }
            }
        }
        return res; 
    }
        
    public boolean isPalindrome(String s) {
        String reversed = new StringBuffer(s).reverse().toString();
        return s.equals(reversed);
    }
}

File: 0005 Longest Palindromic Substring/longest_palindromic_substring_brutal_force.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        count = 0
        res = ""
        length = len(s)
        for i in range(length):
            for j in range(length):
                subs = s[i: j + 1]
                if self.is_palindrome(subs):
                    if len(subs) > count:
                        count = len(subs)
                        res = subs
        return res
        
    def is_palindrome(self, s):
        return s[::-1] == s
    